app/scripts/lib/account-tracker.test.js

```diff
@@ -53,103 +65,344 @@ describe('Account Tracker', () => {
 
-      const newAccounts = accountTracker.store.getState();
+      accountTracker.start();
 
-      const expectedAccounts = {
-        accounts: {
-          [VALID_ADDRESS]: { address: VALID_ADDRESS, balance: UPDATE_BALANCE },
-          [VALID_ADDRESS_TWO]: {
-            address: VALID_ADDRESS_TWO,
-            balance: INITIAL_BALANCE_2,
-          },
-        },
-        currentBlockGasLimit: '',
-      };
+      expect(blockTrackerStub.removeListener).toHaveBeenNthCalledWith(
+        1,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+      expect(blockTrackerStub.addListener).toHaveBeenNthCalledWith(
+        1,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+      expect(updateAccountsWithNetworkClientIdSpy).toHaveBeenNthCalledWith(1); // called first time with no args
 
-      expect(newAccounts).toStrictEqual(expectedAccounts);
+      accountTracker.start();
+
+      expect(blockTrackerStub.removeListener).toHaveBeenNthCalledWith(
+        2,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+      expect(blockTrackerStub.addListener).toHaveBeenNthCalledWith(
+        2,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+      expect(updateAccountsWithNetworkClientIdSpy).toHaveBeenNthCalledWith(2); // called second time with no args
+
+      accountTracker.stop();
     });
+  });
 
-    it('should not change accounts if the passed address is not in accounts', async () => {
-      accountTracker.store.updateState({ accounts: { ...mockAccounts } });
+  describe('stop', () => {
+    it('ends the subscription to the block tracker', async () => {
+      jest.spyOn(blockTrackerStub, 'removeListener').mockImplementation();
 
-      await accountTracker._updateAccount('fake address');
+      accountTracker.stop();
 
-      const newAccounts = accountTracker.store.getState();
+      expect(blockTrackerStub.removeListener).toHaveBeenNthCalledWith(
+        1,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+    });
+  });
+
+  describe('startPollingByNetworkClientId', () => {
+    it('should subscribe to the block tracker and update accounts if not already using the networkClientId', async () => {
+      jest.spyOn(blockTrackerFromHookStub, 'addListener').mockImplementation();
+      const updateAccountsWithNetworkClientIdSpy = jest
+        .spyOn(accountTracker, '_updateAccounts')
+        .mockResolvedValue();
+
+      accountTracker.startPollingByNetworkClientId('mainnet');
+
+      expect(blockTrackerFromHookStub.addListener).toHaveBeenCalledWith(
+        'latest',
+        expect.any(Function),
+      );
+      expect(updateAccountsWithNetworkClientIdSpy).toHaveBeenCalledWith(
+        'mainnet',
+      );
+
+      accountTracker.startPollingByNetworkClientId('mainnet');
+
+      expect(blockTrackerFromHookStub.addListener).toHaveBeenCalledTimes(1);
+      expect(updateAccountsWithNetworkClientIdSpy).toHaveBeenCalledTimes(1);
```

Comment:
the point of this additional call/check is to indicate that we won't  add another listener if we subscribe again for an already active networkClient?

Reply:
exactly